pulsar-transaction/coordinator/src/test/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriterTest.java

```diff
@@ -613,9 +620,322 @@ public ByteBuf serialize(ArrayList<Integer> dataArray) {
+            @Override
+            public void addComplete(Position position, Object context) {
+                addDataCallbackFinishCount.incrementAndGet();
+            }
+            @Override
+            public void addFailed(ManagedLedgerException exception, Object ctx) {
+                addDataCallbackFailureCount.incrementAndGet();
+            }
+        };
+        // Create TxnLogBufferedWriter.
+        OrderedExecutor orderedExecutor =  OrderedExecutor.newBuilder()
+                .numThreads(5).name("txn-threads").build();
+        HashedWheelTimer transactionTimer = new HashedWheelTimer(new DefaultThreadFactory("transaction-timer"),
+                1, TimeUnit.MILLISECONDS);
+        RandomLenSumStrDataSerializer dataSerializer = new RandomLenSumStrDataSerializer();
+        TxnLogBufferedWriter<Integer> txnLogBufferedWriter = new TxnLogBufferedWriter<Integer>(
+                managedLedger, orderedExecutor, transactionTimer,
+                dataSerializer, 256, 1024,
+                1, false, metricsStats);
+        // Add some data.
+        int writeCount = 1000;
+        for (int i = 0; i < writeCount; i++){
+            txnLogBufferedWriter.asyncAddData(1, addDataCallback, "");
+        }
+        // Wait for all data write finish.
+        Awaitility.await().atMost(2, TimeUnit.SECONDS).until(
+                () -> addDataCallbackFinishCount.get() + addDataCallbackFailureCount.get() == writeCount
+        );
+        Assert.assertEquals(addDataCallbackFailureCount.get(), 0);
+        // Assert metrics stat.
+        Assert.assertEquals(getCounterValue(String.format("%s_batched_log_triggering_count_by_force", metricsPrefix)), 0D);
+        Assert.assertEquals(
+                getCounterValue(String.format("%s_batched_log_triggering_count_by_records", metricsPrefix))
+                    + getCounterValue(String.format("%s_batched_log_triggering_count_by_size", metricsPrefix))
+                    + getCounterValue(String.format("%s_batched_log_triggering_count_by_delay_time", metricsPrefix)),
+                0D);
+        Assert.assertEquals(
+                getHistogramCount(String.format("%s_batched_log_records_count_per_entry", metricsPrefix)),
+                0D);
+        Assert.assertEquals(
+                getHistogramCount(String.format("%s_batched_log_entry_size_bytes", metricsPrefix)),
+                0D);
+        Assert.assertEquals(
+                getHistogramCount(String.format("%s_batched_log_oldest_record_delay_time_seconds", metricsPrefix)),
+                0D);
+        // cleanup.
+        txnLogBufferedWriter.close();
+        metricsStats.close();
+        transactionTimer.stop();
+        orderedExecutor.shutdown();
+    }
+
+    /**
+     * Test {@link TxnLogBufferedWriterMetricsStats#triggerFlushByForce(int, long, long)}.
+     */
+    @Test
+    public void testMetricsStatsWhenForceFlush() throws Exception {
```

Comment:
If you remove public force flush then this test can be removed?

Reply:
> If you remove public force flush then this test can be removed?

No, I will rename this method to `testMetricsStatsWhenTriggeredLargeSingleData`